### ISSUE TITLE
[Benchmark] Expand query benchmark to 50–100 specific, multi-constraint queries

### ISSUE THREAD

**Issue by day-yc** [yizhouliwe/EECS545-Project26W#5]
Blocked by: #4
Blocks: #10, #11

The current benchmark has 12 manually labeled queries (only 2 used for validation).
Both GSIs flagged this as the most urgent fix: results over 2 queries are statistically
meaningless, and queries like "semantic retrieval methods" are too broad — TF-IDF's
strong performance on them is not informative.

Requirements:
- Queries must be specific, multi-constraint (describe a method + domain + evaluation setting)
- Bad: "semantic retrieval methods"
- Good: "dense retrieval for low-resource biomedical domains combining contrastive
  pretraining with sparse lexical signals, evaluated on scientific benchmarks"
- Each query needs: query_text, relevant_arxiv_ids (3–10 papers), ground_truth_answer
  (a reference answer generated from the relevant abstracts using the 120B LLM)
- Target: 50–100 queries total, re-split 60/20/20

Tasks:
- [x] Redesign/replace the existing 12 queries with more specific versions
- [x] Write a script src/generate_queries.py that prompts the UM 120B LLM to auto-generate
      specific queries grounded in real papers from the corpus
- [x] For each generated query, have the LLM produce a ground_truth_answer from the
      relevant abstracts (needed for RAGAS evaluation in issue #7)
- [x] Add ground_truth_answer field to labeled_queries.jsonl schema
- [x] Re-run split_queries.py to produce new train/val/test splits
- [x] Verify: relevant set per query should be small (3–10 papers), not 50+

Reference: ASTA benchmark format (https://github.com/allenai/asta-bench)

**Comment by SammiWang0516** [comment]
Hi all, I've resolved this issue by adding one script under src folder called label_with_llm.py to ask UM GPT to label those ground-true answer for each query. Also, I made each query more sophisticated instead of just keywords. Lastly, the evaluation result is shown here but you guys can also use run_part2.py --evaluate to run the result

TFIDF
  Precision@K  0.4421
  Recall@K     0.1957
  NDCG@10      0.4452
  MAP          0.1131

Dense
  Precision@K  0.5105
  Recall@K     0.2334
  NDCG@10      0.5275
  MAP          0.1498

HYBRID
  Precision@K  0.5263
  Recall@K     0.2441
  NDCG@10      0.5515
  MAP          0.1642
